### PR TITLE
Implement `DropDatabase` and `Status`

### LINKED ISSUE #3404: Implement `DropDatabase` and `Status`
### What should be done?

Implement `DropDatabase` and `Status` methods.

### Where?

https://github.com/FerretDB/FerretDB/blob/main/internal/backends/postgresql/backend.go

### Definition of Done

- methods implemented;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #3404.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/backends/postgresql/backend.go [hunk 1]
```diff
@@ -64,8 +64,23 @@ func (b *backend) Name() string {
 
 // Status implements backends.Backend interface.
 func (b *backend) Status(ctx context.Context, params *backends.StatusParams) (*backends.StatusResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/3404
-	return new(backends.StatusResult), nil
+	dbs, err := b.r.DatabaseList(ctx)
```
Comment (chilagrow) [inline internal/backends/postgresql/backend.go:67]: Previous implementation queried `pg_tables` table https://github.com/FerretDB/FerretDB/blob/main/internal/handlers/pg/pgdb/stats.go#L69. This implementation uses metadata from registry.

**Comment by chilagrow** [comment]
Once https://github.com/FerretDB/FerretDB/issues/3406 is merged, integration tests would pass.

```
$ task test-integration-postgresql TEST_RUN='TestCommandsAdministrationCreateDropListDatabases|TestPingCommand|TestCommandsAdministrationServerStatus'
task: [test-integration-postgresql] go test -count=1 -run='TestCommandsAdministrationCreateDropListDatabases|TestPingCommand|TestCommandsAdministrationServerStatus' -timeout=35m -race=true -tags=ferretdb_debug,ferretdb_hana -shuffle=on -coverpkg=../... -coverprofile=integration-postgresql.txt -target-backend=ferretdb-pg -target-tls -postgresql-url='postgres://username@127.0.0.1:5432/ferretdb?pool_max_conns=50&search_path=' -compat-url='mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem' -use-new-pg -disable-filter-pushdown=false -enable-sort-pushdown=false

2023-09-28T14:43:08.673+0900    INFO    setup/startup.go:94     Target system: ferretdb-pg (built-in).
2023-09-28T14:43:08.674+0900    INFO    debug   debug/debug.go:92       Starting debug server on http://127.0.0.1:45535/
2023-09-28T14:43:08.806+0900    INFO    setup/startup.go:105    Compat system: MongoDB (mongodb://username:password@127.0.0.1:47018/?tls=true&tlsCertificateKeyFile=../build/certs/client.pem&tlsCaFile=../build/certs/rootCA-cert.pem).
-test.shuffle 1695879788806711384
PASS
coverage: 16.5% of statements in ../...
ok      github.com/FerretDB/FerretDB/integration        24.164s
```

File: internal/backends/postgresql/backend.go (hunk 1 above)
Comment (AlekSi) [reply internal/backends/postgresql/backend.go:67]: https://github.com/FerretDB/FerretDB/blob/7ca090ff2c5305d5f020bfd4fb555b2175208af7/internal/backends/backend.go#L96-L98

When connection information from `ctx` is already checked, only the cached information is returned.

I think we should get a connection to the first database and call `db.Ping()` on it. If there are no databases – well, it is not that important if PostgreSQL works or not in this case 😆

**Review by AlekSi — CHANGES_REQUESTED** [review]
https://github.com/FerretDB/FerretDB/pull/3451#discussion_r1339816574

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**